### PR TITLE
O3-997: Use formatDate for date and time formatting

### PR DESCRIPTION
**Pull request by Piumal1999**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary

Previously in most of the packages, dayjs was directly used for formatting the date/time values. I replaced those with esm-framework formatDate(), formatTime(), formatDatetime() functions.

## Screenshots

*None.*

## Issue

https://issues.openmrs.org/browse/O3-997

## Other

*None.*

### CODE REVIEW

**Comment by brandones** [comment]
Perfect! It just looks like you need to update `@openmrs/esm-framework`.

**Comment by Piumal1999** [comment]
> It just looks like you need to update `@openmrs/esm-framework`.

Yeah @brandones, could you please provide me with the steps to do that?
I tried updating the esm-framework version in the `package.json`. But still, the formatDate function doesn't operate in the expected behaviour.
image

**Comment by brandones** [comment]
```
yarn upgrade @openmrs/esm-framework openmrs  // to upgrade
git checkout package.json  // to reset the version specifiers to 'next'
```

`package-lock.json` is for NPM. This project uses Yarn. Please read [the docs](http://o3-dev.docs.openmrs.org/#/getting_started/prerequisites?id=javascript-tooling).

**Comment by Piumal1999** [comment]
> ```
> yarn upgrade @openmrs/esm-framework openmrs  // to upgrade
> git checkout package.json  // to reset the version specifiers to 'next'
> ```
> 
> `package-lock.json` is for NPM. This project uses Yarn. Please read [the docs](http://o3-dev.docs.openmrs.org/#/getting_started/prerequisites?id=javascript-tooling).

After doing that, the application throws an error like this:
Screenshot 2022-02-10 at 14 26 11

Screenshot 2022-02-10 at 14 28 03

@brandones, Is it ok if I upgrade all the packages using `yarn upgrade` command?

**Comment by brandones** [comment]
Oh yeah, I don't really know why we've been having an issue with this, but do

```
yarn add -D -W css-loader
```

**Comment by Piumal1999** [comment]
Thanks for the help @brandones. Updating webpack (using `yarn upgrade webpack`) solved the error. Now this pull request is ready for review.

File: packages/esm-generic-patient-widgets-app/src/obs-graph/obs-graph.component.tsx [hunk 1]
```diff
@@ -35,7 +34,7 @@ const ObsGraph: React.FC<ObsGraphProps> = ({ patientUuid }) => {
         .filter((obs) => obs.conceptUuid === selectedConcept.uuid)
         .map((obs) => ({
           group: selectedConcept.label,
-          key: dayjs(obs.issued).format('DD-MMM'),
+          key: formatDate(new Date(obs.issued), { day: false, time: false }),
```
Comment (brandones) [inline packages/esm-generic-patient-widgets-app/src/obs-graph/obs-graph.component.tsx:37]:
```suggestion
          key: formatDate(new Date(obs.issued), { year: false, time: false }),
```

Comment (Piumal1999) [reply packages/esm-generic-patient-widgets-app/src/obs-graph/obs-graph.component.tsx]: Updated with [42bdddf](https://github.com/openmrs/openmrs-esm-patient-chart/pull/539/commits/42bdddfe16f7f8c8fd2adb13bd3ce499c1b77af2)

File: packages/esm-patient-immunizations-app/src/immunizations/immunizations-overview.test.tsx [hunk 2]
```diff
@@ -91,7 +91,7 @@ describe('ImmunizationOverview: ', () => {
       expect(screen.getByRole('columnheader', { name: new RegExp(header, 'i') })).toBeInTheDocument();
     });
 
-    const expectedTableRows = [/rotavirus sep-2018/, /polio nov-2018/, /influenza may-2018/];
+    const expectedTableRows = [/rotavirus sept — 2018/, /polio nov — 2018/, /influenza may — 2018/];
```
Comment (brandones) [inline packages/esm-patient-immunizations-app/src/immunizations/immunizations-overview.test.tsx:94]: I think this one should probably not be `wide`.

**Review by brandones — CHANGES_REQUESTED** [review]
Awesome work @Piumal1999 ! I just have two suggestions and a question, and then this is good to merge.

**Comment by brandones** [comment]
Oh and we'll need to resolve the conflict before merging.

File: packages/esm-patient-test-results-app/src/timeline/timeline.scss [hunk 3]
```diff
@@ -99,7 +99,7 @@
   div {
     position: sticky;
     left: calc(10em + 10px);
-    width: fit-content;
+    width: max-content;
```
[the remark this answers is not in the document]
Comment (Piumal1999) [reply packages/esm-patient-test-results-app/src/timeline/timeline.scss:102]: If we use the `fit-content`, it breaks the date/time (on test-results timeline) into 2 lines and displays like this:

image

When using `max-content`, it adjusts the column length according to the content width.

image

File: packages/esm-patient-vitals-app/src/vitals/vitals-chart.component.tsx [hunk 4]
```diff
@@ -36,7 +36,7 @@ const VitalsChart: React.FC<VitalsChartProps> = ({ patientVitals, conceptUnits,
         return (
           vitals[selectedVitalSign.value] && {
             group: 'vitalsChartData',
-            key: dayjs(vitals.date).format('DD-MMM'),
+            key: formatDate(parseDate(vitals.date.toString()), { mode: 'wide', year: false }),
```
[the remark this answers is not in the document]
Comment (Piumal1999) [reply packages/esm-patient-vitals-app/src/vitals/vitals-chart.component.tsx]: Ok. updated with 8ce9c4e

File: packages/esm-patient-immunizations-app/src/immunizations/immunizations-overview.test.tsx (hunk 2 above)
Comment (Piumal1999) [reply packages/esm-patient-immunizations-app/src/immunizations/immunizations-overview.test.tsx]: Updated with 8ce9c4e

**Comment by Piumal1999** [comment]
Added the request changes and Resolved the conflicts.

**Review by brandones — APPROVED** [review]
This is really helpful. Thanks for this, @Piumal1999 .

**Comment by denniskigen** [comment]
I'm finding that this PR broke a few tests in this repo. Did you get these failures as well, @Piumal1999?

image

**Comment by denniskigen** [comment]
Looks like `formatDate` is being invoked using the wrong signature in some places:

Screenshot 2022-02-15 at 21 41 32

Screenshot 2022-02-15 at 21 40 49

Screenshot 2022-02-15 at 21 38 58

**Comment by denniskigen** [comment]
Oh, apologies! Looks like the function signature changed in https://github.com/openmrs/openmrs-esm-core/pull/305/files and I'm on an old version of `esm-framework` 😅

**Pull request merged**